velero-plugins/pod/restore.go

```diff
@@ -116,11 +117,25 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 		}
 	}
 	// Check if pod has owner Refs and defaultVolumesToRestic flag as false/nil
-	if (len(ownerRefs) > 0 && pod.Annotations[common.ResticBackupAnnotation] == "" && (defaultVolumesToResticFlag == nil || !*defaultVolumesToResticFlag))   && !podHasRestoreHooks {
+	if (len(ownerRefs) > 0 && pod.Annotations[common.ResticBackupAnnotation] == "" && (defaultVolumesToResticFlag == nil || !*defaultVolumesToResticFlag)) && !podHasRestoreHooks {
 		p.Log.Infof("[pod-restore] skipping restore of pod %s, has owner references, no restic backup, and no restore hooks", pod.Name)
 		return velero.NewRestoreItemActionExecuteOutput(input.Item).WithoutRestore(), nil
 	}
 
+	// If pod has both "deployment" and "deploymentconfig" labels, it belongs to a DeploymentConfig
+	// If defaultVolumesToRestic, remove these labels so that the DC won't immediately delete the
```

Comment:
So when does this pod get deleted? never?

So we end up with restored Pod + 1 DeploymentConfig pod?

Reply:
The underlying issue is that if you restore a pod and a DC-not-scaled-to-zero, then the restored pod is immediately deleted. The only way for restic restores or post-restore hooks to work with these pods is to short-circuit that action. There are two ways to do it. The first approach was a docs-only fix -- telling users to first run the restore excluding DCs and ReplicationControllers, then once restic and hooks were done, restore those missing resources from the same backup. The docs-only fix was later rejected in favor of this more active approach, where disconnected pods are restored along with scaled-down DCs. The disconnection was needed to prevent the DC controller from immediately scaling these pods to zero as called for by the modified restored DC. The only other way around this problem is to have users move to Deployments before backup.